File: app/src/main/java/com/beone/bestpractice/MainActivity.java
package com.beone.bestpractice;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;

import com.beone.bestpractice.ui.CountryFragment;
import com.beone.bestpractice.ui.StateFragment;

public class MainActivity extends AppCompatActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    if (savedInstanceState == null) {
      FragmentTransaction fragmentTransaction = this.getSupportFragmentManager().beginTransaction();
      fragmentTransaction.replace(R.id.framelayout, CountryFragment.newInstance(), "state")
              .commit();
    }
  }

  @Override
  protected void onRestoreInstanceState(@NonNull Bundle savedInstanceState) {
    super.onRestoreInstanceState(savedInstanceState);
  }
}

File: app/src/main/java/com/beone/bestpractice/MainActivity.kt
package com.beone.bestpractice

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.beone.bestpractice.ui.CountryFragment

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (savedInstanceState == null) {
            val fragmentTransaction = this.supportFragmentManager.beginTransaction()
            fragmentTransaction.replace(R.id.framelayout, CountryFragment(), "state")
                    .commit()
        }
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
    }
}